drbert/data/dataset_readers.py

```diff
@@ -1,43 +1,67 @@
-import os
 import json
+import logging
+import os
 
 from torchtext import data
 from torchtext import datasets
 
+from ..constants import PARTITIONS
+
+logger = logging.getLogger(__name__)
+
 
 class DatasetReader(object):
     """The parent class for all dataset readers. A user is mean to interact with a subclass of this
-    class (e.g. DocumentClassificationDatasetReader), not this class itself.
+    class (e.g. `DocumentClassificationDatasetReader`), not this class itself.
 
     Args:
         path (str): Common prefix of the splits’ file paths.
-        partitions (dict): A dictionary keyed by partition ('train', 'test', 'validation')
+        partitions (dict): A dictionary keyed by partition(s) ('train', 'validation', 'test')
             containing the suffix to add to `path` for that partition.
         tokenizer (transformers.PreTrainedTokenizer): The function used to tokenize strings into
             sequential examples.
         format (str): Format of the data file. One of “CSV”, “TSV”, or “JSON” (case-insensitive).
-        skip_header (bool) : Optional, whether to skip the first line of the input file. Defaults to
+        skip_header (bool, pptional): Whether to skip the first line of the input file. Defaults to
             False.
-        batch_sizes (tuple): Optional, tuple of batch sizes to use for the different splits, or None
+        batch_sizes (tuple, optional): Tuple of batch sizes to use for the different splits, or None
             to use the same batch_size for all splits. Defaults to None.
-        lower: (bool): Optional, True if text should be lower cased, False if not. Defaults to
+        lower: (bool, optional): True if text should be lower cased, False if not. Defaults to
             False.
         sort_key (function): A key to use for sorting examples in order to batch together examples
             with similar lengths and minimize padding. The sort_key provided to the Iterator
             constructor overrides the sort_key attribute of the Dataset, or defers to it if None.
+        device (str or torch.device, optional): A string or instance of torch.device specifying
+            which device the Tensors are going to be created on. If left as default, the tensors
+            will be created on cpu. Default: None.
 
     Raises:
         ValueError: If batch_sizes is not a tuple or None.
         ValueError: If batch_sizes is not None and `len(batch_sizes) != len(partitions)`.
+        ValueError: If 'train' not `partitions`.
+        ValueError: If any key in `partitions` is not in `drbert.constants.PARTITIONS`.
     """
-    def __init__(self, path, partitions, tokenizer, format, skip_header=False, batch_sizes=None,
-                 lower=False, sort_key=None):
+    def __init__(self, path, partitions, tokenizer, format, skip_header=False,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'format'](https://app.codacy.com/app/JohnGiorgi/drbert/pullRequest?prid=4348433)